### ISSUE TITLE
NO QR code for Wemo Bridge

### ISSUE THREAD

**Issue by jvs2010** [rudders/homebridge-wemo#42]
### What issue do you have? Please be as thorough and explicit as possible.

I was having issues with the latest version of your Wemo Plugin due to my Hoobs version being 4.0. (I was getting an error saying that the plugin may be unstable due to the version of Hoobs I was running.) 

I updated to Hoobs v. 4.1.21 (Homebridge Version 1.3.9), uninstalled the Wemo Bridge and Plugin, cleared the cache of the Hoobs, rebooted my Hoobs and reinstalled the Wemo Bridge and Plug-in. However, when I went to the bridge, I did not have a QR code to scan for Homekit under pairing. (Prior to this, I had a QR code for this bridge.) All of my other bridges have QR codes in the same location.

Furthermore, as soon as the Wemo Bridge was installed (even though I could not link it to my Homekit), I started getting the following error for all of my other bridges:

WARNING unhandled rejection: SyntaxError: Unexpected token u in JSON at position 0

I have since uninstalled the Wemo Bridge and Plug-in due to this error and the instability they were creating with my other bridges.

I would love to fix this, as I was very happy using the Wemo bridge until recently.

### Details of your setup.
* Do you use (1) Homebridge UI-X (2) Homebridge CLI or (3) HOOBS? 

HOOBS

* Which version of Homebridge/HOOBS do you have?

Hoobs 4.1.21

* Which platform do you run Homebridge/HOOBS on (e.g. Raspberry Pi/Windows/HOOBS Box)? Please also mention your version of Node.js/NPM if known.

Hoobs Box
Node Version 16.13.1

* Which version of this plugin (homebridge-wemo) do you have? Has the issue started since upgrading from a previous version?

The latest version (v. 4.5.3) // Yes.

* Which Wemo devices do you have that are causing issues? Please include product models if applicable.

Two switches, two outlets

### Please paste any relevant logs below.

```
1/6/2022, 1:33:37 PMPlatform Orbit BridgeWARNINGunhandled rejection: SyntaxError: Unexpected token u in JSON at position 0
1/6/2022, 1:33:37 PMAugust Smart Locks BridgeWARNINGunhandled rejection: SyntaxError: Unexpected token u in JSON at position 0
1/6/2022, 1:33:37 PMRing BridgeWARNINGunhandled rejection: SyntaxError: Unexpected token u in JSON at position 0
```

**Comment by bwp91** [comment]
Hi @jvs2010 

Can you show me your wemo config? At the bottom of the wemo config page there is a button called "Advanced". Can you click on that and show me the text config?

**Comment by jvs2010** [comment]
I would need to reinstall it. I uninstalled due to the issue it was creating. Give me a moment and I will do that.

**Comment by jvs2010** [comment]
1) I installed the Plugin from the Plugin page. It cycled for a while, created a Wemo Bridge, but when I go to the bridge it says "No plugins installed. Install Plugin."

**Comment by jvs2010** [comment]
2. When I clicked on "Install Plugin" it took me back to the same page. It then gave me the option to install by creating a new bridge or by selecting the "Wemo Bridge" it installed as mentioned in 1 above.

**Comment by jvs2010** [comment]
Sorry, not sure what the "cannot read config file" is in the last comment. I meant in "Number 1" above.

**Comment by jvs2010** [comment]
3. I selected to install it using the Wemo Bridge it created in Number 1 above. I did that and it gave me the uninstall/install/configuration option.

**Comment by jvs2010** [comment]
As soon as I installed the Bridge, I again got these errors for each of my other bridges. I did not even configure the Wemo bridge yet:

1/6/2022, 4:24:00 PMPlatform Orbit BridgeWARNINGunhandled rejection: SyntaxError: Unexpected token u in JSON at position 0
1/6/2022, 4:24:00 PMAugust Smart Locks BridgeWARNINGunhandled rejection: SyntaxError: Unexpected token u in JSON at position 0
1/6/2022, 4:24:00 PMRing BridgeWARNINGunhandled rejection: SyntaxError: Unexpected token u in JSON at position 0

**Comment by jvs2010** [comment]
This is what the config page shows:

{
    "platform": "Wemo"
}

**Comment by jvs2010** [comment]
I configured one light switch:

{
    "platform": "Wemo",
    "name": "Wemo",
    "mode": "semi",
    "wemoLights": [
        {
            "label": "Front Lights",
            "serialNumber": "221533K1300D9B",
            "listenerType": "http"
        }
    ]
}

**Comment by jvs2010** [comment]
But I still do not have the QR code for pairing when I go to the Bridge.

**Comment by bwp91** [comment]
Uninstall the wemo plugin again for now (so the other plugins work again), I think I will ask the HOOBS support staff about this. I will get back to you...

**Comment by jvs2010** [comment]
Thanks.

**Comment by askovi** [comment]
That error comes when something is wrong in the Advanced config. Can you please delete that bridge and recreate it with the wemo plugin only? Paste your config from the Advanced section here to make sure the syntax is correct

**Comment by jvs2010** [comment]
I'm not sure I understand what you mean by "delete that bridge and recreate it with the wemo plugin only"? I already deleted the wemo bridge and plugin, reinstalled it (as mentioned above) and pasted what was in the advanced config section. I got the error a second time, so I once again deleted the bridge and plugin. Do you want me to go through all of the same steps above again?

**Comment by askovi** [comment]
Yes except do not use the advanced config section. Use the plugin configuration page to configure the plugin. It has measures in place to prevent invalid json. Make sure only the wemo plugin is running on a fresh bridge.

**Comment by jvs2010** [comment]
I used the plugin configuration page to configure the plugin last time. (I just copied and pasted the text from the advanced config for Ben so that he could see it. I'm not a coder, so I don't configure the Hoobs using code -- only the web forms.) And it was the only plugin running on the Wemo Bridge it created. (There was nothing else on that bridge.)

**Comment by jvs2010** [comment]
Each time I try to install the Wemo plug-in from the "featured plugins" it creates a bridge, but then tells me there is no plug-in installed. If I click on the hyperlink where it says no plugin is installed, it takes me back to the Wemo plug-in page and "Install" is once again the only choice. It then asks me if I want to create a new bridge or use the Wemo bridge I just created (or another bridge).

**Comment by jvs2010** [comment]
Thanks for your help with this. Would this json error also create the issue where there is no QR code?

**Comment by askovi** [comment]
The QR code only appears when the bridge is running. If the bridge is stopped or in an error state it will not display because it won't be able to communicate with Homekit. 

Not sure what's causing that plugin installation issue for you but there's a fix command you can run from the HOOBS terminal that will check and fix all the issues it can find:

    curl -sk https://dl.hoobs.org/fix | sudo bash -

**Comment by jvs2010** [comment]
Running that now.

**Comment by jvs2010** [comment]
So I ran the fix command you provided. I then installed the Wemo plugin. This time I had no issues with the plugin creating a bridge and then installing. Also, I am no longer getting the json error for my other bridges in the log. HOWEVER, I still don't have a QR code in the plugin section and the bridge only has Start and Cache as options.

**Comment by askovi** [comment]
Click start then see if the code appears, otherwise check the logs. You can download the log file from the top of the log page and post it here

**Comment by jvs2010** [comment]
This is all that happens in the log when I click "start"

```
1/6/2022, 8:20:01 PMwemobridge stopping
1/6/2022, 8:20:06 PMWemo Bridge starting
```

Nothing else happens. (And it still has "Start" and "Cache" as the only options for the bridge.

Is there some reason why there are no spaces for "wemobridge" when it is stopping, but it is spaced out as "Wemo Bridge" when it is starting?

**Comment by askovi** [comment]
I just installed it myself and it runs fine. Is this what you have in the advanced config for wemo? 

![image](https://user-images.githubusercontent.com/3981552/148476422-f04fdcbf-1adb-4d8e-b482-6026ae3c609b.png)

**Comment by jvs2010** [comment]
Here is what I have. I configured this using the configuration page. (I have removed the actual serial numbers and replaced them with Xs in this text file since I am posting it online.)


{
    "accessories": [],
    "platforms": [
        {
            "platform": "Wemo",
            "name": "Wemo",
            "mode": "semi",
            "wemoLights": [
                {
                    "label": "Front Door",
                    "serialNumber": "XXXXX",
                    "listenerType": "http"
                }
            ],
            "wemoOutlets": [
                {
                    "label": "Den Lamp",
                    "serialNumber": "XXXXXXX",
                    "showAs": "default",
                    "listenerType": "http"
                }
            ]
        }
    ]
}

**Comment by jvs2010** [comment]
When I go to Homekit, if I choose add an accessory and click "more options", I see "Wemo Bridge FA26" as a choice.

**Comment by askovi** [comment]
that json looks valid. you see it but you can't add it cause you dont have the code. Have you tried rebooting the HOOBS Box? top right menu in hub settings. Also are you using the HOOBS Desktop app?

**Comment by jvs2010** [comment]
I rebooted the HOOBS box earlier, but i can try again. I didn't know about a Desktop app. I've been accessing HOOBS via hoobs.local

**Comment by jvs2010** [comment]
After rebooting the HOOBS box, the QR code appeared for the Wemo Bridge. I was able to add it to my Homekit. But... no Wemo accessories were added to my homekit. When I checked the log, it said:
No plugins installed.
1/6/2022, 8:49:59 PMWemo BridgeLoading 1 platforms...
1/6/2022, 8:49:59 PMWemo BridgeBridge is running on port 51856.

So then I had to reinstall the Wemo Plugin.

I restarted both the plugin and the bridge.

Once I did that, it added one of the lights I had configured -- but gave it the wrong name! And it didn't add the switch I had configured. I am going to try uninstalling and reinstalling again.

**Comment by jvs2010** [comment]
After clearing all of the caches (for the Hub and the plugin), deleting the bridge, rebooting the HOOBS again and reinstalling the Wemo plugin again... I finally got it to add my other switches and outlets, but it still insists on giving two switches with different serial numbers the same name (an old name I used in a previous iteration, but not the name I gave either switch in the configuration). I went ahead and added the bridge to Homekit using the QR code and renamed the lights in Homekit. Everything (fingers crossed) now appears to be working. Many thanks for your help, @askovi. I really appreciate it.
